### ISSUE TITLE
Ruby LSP stays forever in `Starting Ruby LSP...`

### ISSUE THREAD

**Issue by tachyons** [Shopify/ruby-lsp#1157]
### Ruby version

3.1.4

### Code snippet

```
➜ ruby-lsp
Ruby LSP> Skipping custom bundle setup since /Users/apple/code/gitlab_client/.ruby-lsp/Gemfile.lock already exists and is up to date
Ruby LSP> Running bundle install for the custom bundle. This may take a while...
Ruby LSP> Command: (bundle check || bundle install) 1>&2
The Gemfile's dependencies are satisfied
Starting Ruby LSP...
```

### Description

ruby-lip takes forever in starting mode. It used to work before, But after updating to latest version, it stays in starting phase. 

```
^C/Users/user/.asdf/installs/ruby/3.1.4/lib/ruby/gems/3.1.0/gems/language_server-protocol-3.17.0.3/lib/language_server/protocol/transport/io/reader.rb:16:in `gets': Interrupt
        from /Users/user/.asdf/installs/ruby/3.1.4/lib/ruby/gems/3.1.0/gems/language_server-protocol-3.17.0.3/lib/language_server/protocol/transport/io/reader.rb:16:in `read'
        from /Users/user/.asdf/installs/ruby/3.1.4/lib/ruby/gems/3.1.0/gems/ruby-lsp-0.12.0/lib/ruby_lsp/server.rb:65:in `start'
        from /Users/user/.asdf/installs/ruby/3.1.4/lib/ruby/gems/3.1.0/gems/sorbet-runtime-0.5.11089/lib/types/private/methods/_methods.rb:277:in `bind_call'
        from /Users/user/.asdf/installs/ruby/3.1.4/lib/ruby/gems/3.1.0/gems/sorbet-runtime-0.5.11089/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
```

### Expected output

ruby-lsp starts without any issues.

**Comment by vinistock** [comment]
Thank you for the report! Based on the backtrace, it looks like the LSP booted properly given that it's sitting in the main loop reading input from STDIN. If the LSP never receives any requests from the editor, it'll always sit forever in the infinite loop waiting for something to be written to STDIN.

Can you share which editor and what configurations you are using? It seems that for some reason requests are not being made to the LSP.

If you ran the `ruby-lsp` in the terminal, then you're seeing the correct behaviour. We print `Starting the Ruby LSP` when we first start booting and then print that we finished initializing only once we received the `initialized` notification from the editor.

**Comment by tachyons** [comment]
Just tested on neovim, it is working. Could be vs code specific issue, how do I find logs when the vscode executes the ruby-lsp ?

**Comment by tsvallender** [comment]
I think [my issue](https://github.com/Shopify/ruby-lsp/issues/1158) is unrelated sorry.

**Comment by vinistock** [comment]
@tachyons do you have anything printed in the `Ruby LSP` channel in VS Code's `Output` tab?

**Comment by tachyons** [comment]
TIL about output tab. This was the error. 

```
The language client requires VS Code version ^1.82.0 but received version 1.81.1
```

**Comment by tachyons** [comment]
There is still scope for improvement by showing the error on vscode extension UI so that the users can identify and fix the issue

**Comment by vinistock** [comment]
@dderyldowney do you have [this message](https://github.com/Shopify/ruby-lsp/issues/1157#issuecomment-1788254451) in the `Output` tab?

If you do, then you need to update VS Code. If you don't, then your issue is not the same one. Can you please open a separate issue and include if anything got printed to the output tab in the report?
